cmake with DTFLITE_KERNEL_TEST=on in ubuntu reports errors

Platform: Ubuntu 20.04
Steps to reproduce:
1. git clone https://github.com/tensorflow/tensorflow
2. mkdir tflite_build & cd tflite_build
3. cmake ../tensorflow/tensorflow/lite -DCMAKE_BUILD_TYPE=Debug -DTFLITE_KERNEL_TEST=on -DTFLITE_ENABLE_GPU=ON
Below errors are printed:
```
CMake Error at kernels/CMakeLists.txt:145 (add_executable):
Cannot find source file:



random_uniform_test.cc



Tried extensions .c .C .c++ .cc .cpp .cxx .cu .m .M .mm .h .hh .h++ .hm
.hpp .hxx .in .txx
Call Stack (most recent call first):
kernels/CMakeLists.txt:312 (add_kernel_test)




CMake Error at kernels/CMakeLists.txt:145 (add_executable):
Cannot find source file:



random_standard_normal_test.cc



Tried extensions .c .C .c++ .cc .cpp .cxx .cu .m .M .mm .h .hh .h++ .hm
.hpp .hxx .in .txx
Call Stack (most recent call first):
kernels/CMakeLists.txt:312 (add_kernel_test)




CMake Error at kernels/CMakeLists.txt:145 (add_executable):
No SOURCES given to target: random_uniform_test
Call Stack (most recent call first):
kernels/CMakeLists.txt:312 (add_kernel_test)




CMake Error at kernels/CMakeLists.txt:145 (add_executable):
No SOURCES given to target: random_standard_normal_test
Call Stack (most recent call first):
kernels/CMakeLists.txt:312 (add_kernel_test)




CMake Generate step failed. Build files cannot be regenerated correctly.
```
If I do below changes, it works:
```
diff --git a/tensorflow/lite/kernels/CMakeLists.txt b/tensorflow/lite/kernels/CMakeLists.txt
index daa390fa93c..67ccb896896 100644
--- a/tensorflow/lite/kernels/CMakeLists.txt
+++ b/tensorflow/lite/kernels/CMakeLists.txt
@@ -252,8 +252,6 @@ set(TEST_WITH_EXTERNAL_MAIN_LIST
   pow_test.cc
   quant_basic_lstm_test.cc
   quantize_test.cc
-  random_standard_normal_test.cc
-  random_uniform_test.cc
   range_test.cc
   rank_test.cc
   reduce_test.cc
@@ -323,4 +321,4 @@ if(${CMAKE_CROSSCOMPILING})
     ${CMAKE_CURRENT_BINARY_DIR}/run-tests.cmake
     COPYONLY
   )
```
However, in the next build step `cmake --build . -j8`. I will meet many build failures.

And I am a newbie for tflite gpu. I want to be familiar with tflite gpu gl backend. So could you help me how to run the kernel tests on ubuntu with gpu-gl backend? Which target should I build?

> Hi, did you make changes in `CMakesLists.txt` from `random_standard_normal_test.cc` `random_uniform_test.cc` to `random_ops_test.cc`, since in future the existing tests for random ops will be handled in `random_ops_test`.

Yes, I did the changes. However, I still meet above build errors.
The steps to reproduce:
1. Do below changes:
```
--- a/tensorflow/lite/kernels/CMakeLists.txt
+++ b/tensorflow/lite/kernels/CMakeLists.txt
@@ -252,8 +252,7 @@ set(TEST_WITH_EXTERNAL_MAIN_LIST
   pow_test.cc
   quant_basic_lstm_test.cc
   quantize_test.cc
-  random_standard_normal_test.cc
-  random_uniform_test.cc
+  random_ops_test.cc
   range_test.cc
   rank_test.cc
   reduce_test.cc
@@ -323,4 +322,4 @@ if(${CMAKE_CROSSCOMPILING})
     ${CMAKE_CURRENT_BINARY_DIR}/run-tests.cmake
     COPYONLY
   )
-endif()
```
2. cmake ../tensorflow/tensorflow/lite -DCMAKE_BUILD_TYPE=Debug -DTFLITE_KERNEL_TEST=on -DTFLITE_ENABLE_GPU=ON
3. cmake --build . -j8 -t conv_test
Then, below errors are shown:
```
[100%] Linking CXX executable conv_test
/usr/bin/ld: ../libtensorflow-lite.a(op_signature.cc.o): in function `tflite::GetOpSignature(tflite::OperatorCode const*, tflite::Operator const*, tflite::SubGraph const*, tflite::Model const*)':
/home/wp/workspace/jiajia/tensorflow/tensorflow/lite/tools/versioning/op_signature.cc:113: multiple definition of `tflite::GetOpSignature(tflite::OperatorCode const*, tflite::Operator const*, tflite::SubGraph const*, tflite::Model const*)'; libtensorflow-lite-test-base.a(op_signature.cc.o):/home/wp/workspace/jiajia/tensorflow/tensorflow/lite/tools/versioning/op_signature.cc:113: first defined here
/usr/bin/ld: ../libtensorflow-lite.a(op_signature.cc.o): in function `tflite::GetOpSignature(TfLiteContext const*, TfLiteNode const*, TfLiteRegistration const*)':
/home/wp/workspace/jiajia/tensorflow/tensorflow/lite/tools/versioning/op_signature.cc:233: multiple definition of `tflite::GetOpSignature(TfLiteContext const*, TfLiteNode const*, TfLiteRegistration const*)'; libtensorflow-lite-test-base.a(op_signature.cc.o):/home/wp/workspace/jiajia/tensorflow/tensorflow/lite/tools/versioning/op_signature.cc:233: first defined here
/usr/bin/ld: ../libtensorflow-lite.a(external_delegate.cc.o): in function `TfLiteExternalDelegateOptionsInsert(TfLiteExternalDelegateOptions*, char const*, char const*)':
/home/wp/workspace/jiajia/tensorflow/tensorflow/lite/delegates/external/external_delegate.cc:183: multiple definition of `TfLiteExternalDelegateOptionsInsert(TfLiteExternalDelegateOptions*, char const*, char const*)'; libtensorflow-lite-test-base.a(external_delegate.cc.o):/home/wp/workspace/jiajia/tensorflow/tensorflow/lite/delegates/external/external_delegate.cc:183: first defined here
/usr/bin/ld: ../libtensorflow-lite.a(external_delegate.cc.o): in function `TfLiteExternalDelegateOptionsDefault':
/home/wp/workspace/jiajia/tensorflow/tensorflow/lite/delegates/external/external_delegate.cc:194: multiple definition of `TfLiteExternalDelegateOptionsDefault'; libtensorflow-lite-test-base.a(external_delegate.cc.o):/home/wp/workspace/jiajia/tensorflow/tensorflow/lite/delegates/external/external_delegate.cc:194: first defined here
/usr/bin/ld: ../libtensorflow-lite.a(external_delegate.cc.o): in function `TfLiteExternalDelegateCreate':
/home/wp/workspace/jiajia/tensorflow/tensorflow/lite/delegates/external/external_delegate.cc:206: multiple definition of `TfLiteExternalDelegateCreate'; libtensorflow-lite-test-base.a(external_delegate.cc.o):/home/wp/workspace/jiajia/tensorflow/tensorflow/lite/delegates/external/external_delegate.cc:206: first defined here
/usr/bin/ld: ../libtensorflow-lite.a(external_delegate.cc.o): in function `TfLiteExternalDelegateDelete':
/home/wp/workspace/jiajia/tensorflow/tensorflow/lite/delegates/external/external_delegate.cc:215: multiple definition of `TfLiteExternalDelegateDelete'; libtensorflow-lite-test-base.a(external_delegate.cc.o):/home/wp/workspace/jiajia/tensorflow/tensorflow/lite/delegates/external/external_delegate.cc:215: first defined here
collect2: error: ld returned 1 exit status
make[3]: *** [kernels/CMakeFiles/conv_test.dir/build.make:172: kernels/conv_test] Error 1
make[2]: *** [CMakeFiles/Makefile2:11521: kernels/CMakeFiles/conv_test.dir/all] Error 2
make[1]: *** [CMakeFiles/Makefile2:11528: kernels/CMakeFiles/conv_test.dir/rule] Error 2
make: *** [Makefile:2472: conv_test] Error 2

--- a/tensorflow/lite/kernels/CMakeLists.txt
+++ b/tensorflow/lite/kernels/CMakeLists.txt
@@ -252,8 +252,7 @@ set(TEST_WITH_EXTERNAL_MAIN_LIST
   pow_test.cc
   quant_basic_lstm_test.cc
   quantize_test.cc
-  random_standard_normal_test.cc
-  random_uniform_test.cc
+  random_ops_test.cc
   range_test.cc
   rank_test.cc
   reduce_test.cc

